src/core/src/package_managers/YumPackageManager.py

```diff
@@ -345,28 +371,331 @@ def get_package_size(self, output):
+        except Exception as error:
+            error_message = "Exception during fetching and logging default auto update settings on the machine. [Exception={0}]".format(repr(error))
+            self.composite_logger.log_error(error_message)
+            self.status_handler.add_error_to_status(error_message, Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
+            raise
+
+    def is_image_default_patch_configuration_backup_valid(self, image_default_patch_configuration_backup):
+        if self.download_updates_identifier_text in image_default_patch_configuration_backup and self.apply_updates_identifier_text in image_default_patch_configuration_backup and self.enable_on_reboot_identifier_text in image_default_patch_configuration_backup:
+            self.composite_logger.log_debug("Extension already has a valid backup of the default system configuration settings for auto OS updates.")
+            return True
+        else:
+            self.composite_logger.log_error("Extension does not have a valid backup of the default system configuration settings for auto OS updates.")
+            return False
+
+    def __get_current_auto_os_updates_setting_on_machine(self):
+        """ Gets all the update settings related to auto OS updates currently set on the machine """
+        try:
+            self.enable_on_reboot_value = self.is_service_set_to_enable_on_reboot(self.enable_on_reboot_check_cmd)
+
+            self.composite_logger.log_debug("Checking if auto updates are currently enabled...")
+            image_default_patch_configuration = self.env_layer.file_system.read_with_retry(self.os_patch_configuration_settings_file_path, raise_if_not_found=False)
+            if image_default_patch_configuration is not None:
+                settings = image_default_patch_configuration.strip().split('\n')
+                for setting in settings:
+                    match = re.search(self.download_updates_identifier_text + self.auto_update_config_pattern_match_text, str(setting))
+                    if match is not None:
+                        self.download_updates_value = match.group(1)
+
+                    match = re.search(self.apply_updates_identifier_text + self.auto_update_config_pattern_match_text, str(setting))
+                    if match is not None:
+                        self.apply_updates_value = match.group(1)
+
+            if self.download_updates_value == "":
+                self.composite_logger.log_debug("Machine did not have any value set for [Setting={0}]".format(str(self.download_updates_identifier_text)))
+
+            if self.apply_updates_value == "":
+                self.composite_logger.log_debug("Machine did not have any value set for [Setting={0}]".format(str(self.apply_updates_identifier_text)))
+
+            elif self.apply_updates_value == "yes":
```

Comment:
Why is this check in some places with .lower() and some places like here without it?

It feels like all writes and reads to this property needs to go through an internal read and write function, with the property being fully internal.

Reply:
Resolved